src/tiled/preferences.cpp

```diff
@@ -802,8 +807,28 @@ void Preferences::setObjectTypesFile(const QString &fileName)
     if (mObjectTypesFile == fileName)
         return;
 
+    if (!mObjectTypesFile.isEmpty())
+        mWatcher.removePath(mObjectTypesFile);
+
     mObjectTypesFile = fileName;
     mSettings->setValue(QLatin1String("Storage/ObjectTypesFile"), fileName);
 
+    mWatcher.addPath(mObjectTypesFile);
+
     emit stampsDirectoryChanged(fileName);
 }
+
+void Preferences::objectTypesFileChangedOnDisk(const QString &fileName)
+{
+    ObjectTypesSerializer objectTypesSerializer;
+    ObjectTypes objectTypes;
+    bool success = objectTypesSerializer.readObjectTypes(objectTypesFile(), objectTypes);
+
+    if (!fileName.isEmpty())
+        mWatcher.removePath(fileName);
```

Comment:
Why remove the path and then add it again? Also, won't `!fileName.isEmpty()` always be `true` since `fileChanged` is not going to be emitted for empty paths?

Reply:
I agree. This was something left over when I was figuring out how to prevent the object type editor from writing again if it caused the emit. I ended up having the object type editor handle ignoring the signal making this part unnecessary.